angular-chess/src/app/services/move-execution.service.ts

```diff
@@ -25,6 +25,7 @@ export class MoveExecutionService {
   public executeMove(move: Move): void {
     console.log("executeMove: " + JSON.stringify(move));
 
+    move.board = JSON.parse(JSON.stringify(this.boardService.getBoard()));
```

Comment:
Why need this double convert?:
`
 JSON.parse(JSON.stringify(
`

Please remove this.

Reply:
It's used for creating a deep copy of the board. Please see https://stackoverflow.com/questions/36124363/deep-copying-objects-in-angular for details.

Reply:
Idea: Extract this deep copy to an extra function in a util lib.

Reply:
done :-)